core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.controller.spec.ts

```diff
@@ -64,138 +35,244 @@ describe('Translation Suggestion Review Modal Controller', function() {
+        $uibModalInstance: $uibModalInstance,
+        initialSuggestionId: 'suggestion_1',
+        reviewable: reviewable,
+        suggestionIdToSuggestion: angular.copy(suggestionIdToSuggestion)
+      });
+    }));
+
+    it('should initialize $scope properties after controller is initialized',
+      function() {
+        expect($scope.activeSuggestionId).toBe('suggestion_1');
+        expect($scope.activeSuggestion).toEqual(suggestion1);
+        expect($scope.reviewable).toBe(reviewable);
+        expect($scope.reviewMessage).toBe('');
+      });
+
+    it('should register Contributor Dashboard view suggestion for review ' +
+      'event after controller is initialized', function() {
+      expect(
+        SiteAnalyticsService
+          .registerContributorDashboardViewSuggestionForReview)
+        .toHaveBeenCalledWith('Translation');
     });
-  }));
 
-  it('should initialize $scope properties after controller is initialized',
-    function() {
+    it('should accept suggestion in suggestion modal service when clicking on' +
+      ' accept and review next suggestion button', function() {
       expect($scope.activeSuggestionId).toBe('suggestion_1');
       expect($scope.activeSuggestion).toEqual(suggestion1);
       expect($scope.reviewable).toBe(reviewable);
       expect($scope.reviewMessage).toBe('');
+
+      spyOn(
+        SiteAnalyticsService,
+        'registerContributorDashboardAcceptSuggestion');
+      spyOn(contributionAndReviewService, 'resolveSuggestionToExploration')
+        .and.callFake((
+            targetId, suggestionId, action, reviewMessage, commitMessage,
+            callback) => {
+          callback();
+        });
+
+      $scope.reviewMessage = 'Review message example';
+      $scope.acceptAndReviewNext();
+
+      expect($scope.activeSuggestionId).toBe('suggestion_2');
+      expect($scope.activeSuggestion).toEqual(suggestion2);
+      expect($scope.reviewable).toBe(reviewable);
+      expect($scope.reviewMessage).toBe('');
+      expect(
+        SiteAnalyticsService.registerContributorDashboardAcceptSuggestion)
+        .toHaveBeenCalledWith('Translation');
+      expect(contributionAndReviewService.resolveSuggestionToExploration)
+        .toHaveBeenCalledWith(
+          '1', 'suggestion_1', 'accept', 'Review message example',
+          'hint section of "StateName" card', $scope.showNextItemToReview);
+
+      $scope.reviewMessage = 'Review message example 2';
+      $scope.acceptAndReviewNext();
+      expect(
+        SiteAnalyticsService.registerContributorDashboardAcceptSuggestion)
+        .toHaveBeenCalledWith('Translation');
+
+      expect(contributionAndReviewService.resolveSuggestionToExploration)
+        .toHaveBeenCalledWith(
+          '2', 'suggestion_2', 'accept', 'Review message example 2',
+          'hint section of "StateName" card', $scope.showNextItemToReview);
+      expect($uibModalInstance.close).toHaveBeenCalledWith([
+        'suggestion_1', 'suggestion_2']);
```

Comment:
Can you add a comment before each "stage" of this test? Seems like this test has 3 stages:

1. Pre-review. Validate active suggestion.
2. Accept suggestion 1
3. Accept suggestion 2

Ditto for the tests below.

Reply:
These are old test block, I've just added them to a block. I'll create an issue to fix this.